test/arcade/patch/agent/process/PatchProcessInflammationCD4Test.java

```diff
@@ -0,0 +1,179 @@
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import ec.util.MersenneTwisterFast;
+import arcade.core.sim.Simulation;
+import arcade.core.util.Parameters;
+import arcade.patch.agent.cell.PatchCellCART;
+import arcade.patch.env.lattice.PatchLattice;
+import arcade.patch.env.location.PatchLocation;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+import static arcade.core.ARCADETestUtilities.randomDoubleBetween;
+
+public class PatchProcessInflammationCD4Test {
+
+    private PatchProcessInflammationCD4 inflammation;
+
+    private PatchCellCART mockCell;
+
+    private PatchLattice mockLattice;
+
+    private Parameters mockParameters;
+
+    private Simulation mockSim;
+
+    private MersenneTwisterFast mockRandom;
+
+    private double cellVolume;
+
+    private PatchLocation mockLocation;
+
+    @BeforeEach
+    public final void setUp() {
+        mockCell = Mockito.mock(PatchCellCART.class);
+        mockParameters = Mockito.mock(Parameters.class);
+        mockSim = Mockito.mock(Simulation.class);
+        mockRandom = Mockito.mock(MersenneTwisterFast.class);
+        mockLocation = mock(PatchLocation.class);
+        mockLattice = mock(PatchLattice.class);
+
+        Mockito.when(mockCell.getParameters()).thenReturn(mockParameters);
+        cellVolume = randomDoubleBetween(165, 180);
+        when(mockCell.getVolume()).thenReturn(cellVolume);
+        when(mockCell.getLocation()).thenReturn(mockLocation);
+        when(mockLocation.getVolume()).thenReturn(3.0 / 2.0 / Math.sqrt(3.0) * 30 * 30 * 8.7);
+        when(mockParameters.getDouble(anyString())).thenReturn(1.0);
+        when(mockParameters.getInt(anyString())).thenReturn(1);
+
+        when(mockSim.getLattice(anyString())).thenReturn(mockLattice);
+        doNothing().when(mockLattice).setValue(any(PatchLocation.class), anyDouble());
+    }
+
+    @Test
+    public void constructor_called_setsParameters()
+            throws NoSuchFieldException, IllegalAccessException {
+        inflammation = new PatchProcessInflammationCD4(mockCell);
+        assertNotNull(inflammation);
+
+        Field il2SynthesisDelay =
+                PatchProcessInflammationCD4.class.getDeclaredField("iL2SynthesisDelay");
+        il2SynthesisDelay.setAccessible(true);
+        assertEquals(1, il2SynthesisDelay.get(inflammation));
+    }
+
+    @Test
+    public void stepProcess_called_updatesEnvironment()
+            throws NoSuchFieldException, IllegalAccessException {
+        inflammation = new PatchProcessInflammationCD4(mockCell);
+        inflammation.active = true;
+        inflammation.activeTicker = 10;
+        inflammation.iL2Ticker = 10;
+        inflammation.boundArray = new double[180];
+        Arrays.fill(inflammation.boundArray, 10000);
+
+        inflammation.stepProcess(mockRandom, mockSim);
+
+        verify(mockLattice, times(1)).setValue(any(PatchLocation.class), eq(2.4552984483463867E13));
```

Comment:
This value is way too specific (and seems very large? like problematically large)